### PR TITLE
Move kubetest utils to separate package

### PR DESCRIPTION
**Pull request by Q-Lee**
Splitting kubetest's util into separate packages.

1. Generic, stateless functions go to kubetest/util
1. Functions governing kubetest exec/command/process state go to kubetest/process
1. Remaining functions remain in kubetest/util.go

This is a necessary step before placing any provider in its own package, because they all depend on util. This would create a circular depending (e.g., gke > main > gke).

### CODE REVIEW

**Comment by krzyzacy** [comment]
thanks for doing this!

**Comment by krzyzacy** [comment]
@Q-Lee assume there's no new functionality, and this just refactors control and util? github diff is awful :-(

**Comment by BenTheElder** [comment]
/hold
will finish kubetest bump after lunch, push that, review this, unhold

**Comment by krzyzacy** [comment]
/lgtm
punt to @BenTheElder to merge and bump

**Comment by Q-Lee** [comment]
@krzyzacy it's just a refactor. My other kubetest work lives in subpackages, which is only possible if we break out common utilities. In the long run, it might be nice to break more of this out.

**Comment by BenTheElder** [comment]
Sleep() in unit tests :-1:

**Comment by BenTheElder** [comment]
/hold
`TestOutputOutputs` is :-1:

File: kubetest/process/process_test.go [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+		if tc.causeTermination {
+			terminate.Reset(0)
+		}
+		var cmd *exec.Cmd
+		if !tc.pass {
+			cmd = exec.Command("false")
+		} else if tc.sleep == 0 {
+			cmd = exec.Command("true")
+		} else {
+			cmd = exec.Command("sleep", strconv.Itoa(tc.sleep))
```
Comment (BenTheElder) [inline kubetest/process/process_test.go]: @krzyzacy why are we shelling out to `sleep` here?

**Comment by BenTheElder** [comment]
/hold cancel
/retest
We're going to handle the tests in a follow up.

File: kubetest/process/process_test.go (hunk 1 above)
Comment (BenTheElder) [reply kubetest/process/process_test.go]: I looked some more, we have our own custom version of cmd.Output essentially so we can timeout the process potentially. I think we can write a more straightforward version but that's beside the point re: solving the new failures here.

**Comment by BenTheElder** [comment]
I can reproduce this failure locally on my workstation

**Comment by BenTheElder** [comment]
This should fix it:
```diff
diff --git a/kubetest/process/process_test.go b/kubetest/process/process_test.go
index a428dc0a1..0d80d5963 100644
--- a/kubetest/process/process_test.go
+++ b/kubetest/process/process_test.go
@@ -355,9 +355,9 @@ func TestFinishRunningParallel(t *testing.T) {
 }

 func TestOutputOutputs(t *testing.T) {
-       interrupt := time.NewTimer(time.Duration(0))
-       terminate := time.NewTimer(time.Duration(0))
-       c := NewControl(time.Duration(0), interrupt, terminate, false)
+       interrupt := time.NewTimer(time.Second * 1)
+       terminate := time.NewTimer(time.Second * 1)
+       c := NewControl(time.Second * 1, interrupt, terminate, false)

        b, err := c.Output(exec.Command("echo", "hello world"))
        txt := string(b)
```

File: kubetest/process/process_test.go [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+			t.Errorf("TC %q did not reset the terminate timer: %v", tc.name, err)
+		}
+		if tc.causeTermination && !c.terminated {
+			t.Errorf("TC %q did not terminate, err: %v", tc.name, err)
+		}
+	}
+}
+
+func TestOutputOutputs(t *testing.T) {
+	interrupt := time.NewTimer(time.Duration(20) * time.Millisecond)
```
Comment (BenTheElder) [inline kubetest/process/process_test.go]: this is probably too low, also time.Duration(20) is unnecessary, just do `time.Second * 1` or similar

**Comment by krzyzacy** [comment]
what's wrong?

**Comment by ixdy** [comment]
PR gce e2e jobs are failing with
```
W0227 01:21:55.465] 2018/02/27 01:21:55 main.go:278: Something went wrong: failed to acquire k8s binaries: open /go/src/k8s.io/kubernetes/_control.Output/gcs-stage: no such file or directory
```

**Comment by ixdy** [comment]
(I'm working on a hotfix)

**Pull request merged**